processor/aggregationprocessor/README.md

```diff
@@ -13,17 +13,17 @@ This processor aggregates metrics over a configurable interval, allowing for met
 3. If the metric name does not match the `include` regex, the metric passes through the processor.
 4. If the metric matches, but is not a gauge or cumulative sum, the metric passes through the processor.
 5. If the metric name does match, and the metric is a gauge or cumulative sum, the metric is added to an aggregate based on its attributes. The metric does not continue down the pipeline.
-6. After the configured `interval` has passed, all aggregate metrics are emitted. Aggregate metrics are emitted with a name based on the configured `metric_name` for the aggregation.
+6. After the configured `interval` has passed, all aggregate metrics are emitted. Aggregate metrics are emitted with a name based on the configured `metric_name_expression` for the aggregation.
 7. All aggregations are cleared, and will not be emitted on the next interval, unless another matching metric enters the pipeline.
 
 ## Configuration
 | Field                        | Type           | Default  | Description                                                                                                                      |
 |------------------------------|----------------|----------|----------------------------------------------------------------------------------------------------------------------------------|
 | `interval`                   | duration       | `1m`     | The interval on which to emit aggregate metrics.                                                                                 |
-| `include`                    | regex          | `"^.*$"` | A regex that specifies which metrics to consider for aggregation. The default regex matches all metrics.                         |
-| `aggregations`               | []map          | `[{type: min, metric_name: "$$0.min"}, {type: max, metric_name: "$$0.max"}, {type: avg, metric_name: "$$0.avg"}]`| A list of aggregations to perform on each metric.                                                                                |
+| `include`                    | regex          | `".*"` | A regex that specifies which metrics to consider for aggregation. The default regex matches all metrics.                         |
+| `aggregations`               | []map          | `[{type: min, metric_name_expression: 'metric_name + ".min"'}, {type: max, metric_name_expression: 'metric_name + ".max"'}, {type: avg, metric_name_expression: 'metric_name + ".avg"'}]`| A list of aggregations to perform on each metric.                                                                                |
```

Comment:
I missed the discussion on this but I feel like embedding a match expression is a bit complicated. I wasn't sure at first how I was suppose to configure this. I believe the intent is to give the user flexibility to name their aggregate metrics. I'm just wondering how many we really think will need that flexibility. I almost think if you really need that you can just chain a transform processor to rename the aggregate metrics and simplify configuration here where we just always do `metric_name.min`.

Reply:
Good point; I've reworked to remove custom templates for metric names.